### PR TITLE
Move Dataset to an isolated model type

### PR DESCRIPTION
**Pull request by brianhelba**

### CODE REVIEW

File: web_external/Datasets/datasetPage.pug [hunk 1]
```diff
@@ -32,9 +32,6 @@
       tr
         td.isic-listing-item-table-label Owner
         td= dataset.get('owner')
-      tr
-        td.isic-listing-item-table-label Signature
-        td= dataset.get('signature')
   if dataset.canAdmin()
```
Comment (msmolens) [inline web_external/Datasets/datasetPage.pug]: Do these permission checks need to be updated? It looks like being a Dataset Contributor is insufficient to access the metadata endpoints.

Comment (brianhelba) [reply web_external/Datasets/datasetPage.pug]: Fixed.

File: web_external/Datasets/datasetInfoWidget.pug [hunk 2]
```diff
@@ -11,6 +11,13 @@ table.table.table-condensed.isic-dataset-info-table
     tr
       td.isic-dataset-info-table-label License
       td= dataset.get('license')
+    tr
+      td.isic-dataset-info-table-label Attribution
+      td= dataset.get('attribution')
```
Comment (msmolens) [inline web_external/Datasets/datasetInfoWidget.pug]: Attribution isn't available for the dataset summary, so displaying it should be inside a conditional.

File: web_external/Datasets/datasetInfoWidget.pug [hunk 3]
```diff
@@ -11,6 +11,13 @@ table.table.table-condensed.isic-dataset-info-table
...
+    if dataset.canWrite()
+      tr
+        td.isic-dataset-info-table-label Owner
+        td= dataset.get('owner')
```
Comment (msmolens) [inline web_external/Datasets/datasetInfoWidget.pug]: Same here.

File: web_external/Datasets/selectDatasetView.js [hunk 4]
```diff
@@ -21,7 +21,10 @@ const SelectDatasetView = View.extend({
     datasetChanged: function () {
         const datasetId = this.$('select').val();
         const dataset = this.collection.get(datasetId);
-        this.trigger('changed', dataset);
+        dataset.fetch()
+            .done(() => {
+                this.trigger('changed', dataset);
+            });
```
Comment (msmolens) [inline web_external/Datasets/selectDatasetView.js]: Why is it necessary to fetch the dataset in this view?

Comment (msmolens) [reply web_external/Datasets/selectDatasetView.js]: I see, so the details are available!

**Pull request merged**